### PR TITLE
Download binary if it doesn't exist already

### PR DESCRIPTION
**Pull request by riya-n**

### CODE REVIEW

File: script.sh [hunk 1]
```diff
@@ -29,7 +29,9 @@ TOKEN=${INPUT_TOKEN:-${TRUNK_API_TOKEN}} # Defaults to TRUNK_API_TOKEN env var.
 
 # CLI.
 set -x
-curl -fsSL --retry 3 "https://trunk.io/releases/analytics-cli/latest" -o ./trunk-analytics-uploader
+if [[ ! (-f ./trunk-analytics-uploader) ]]; then
```
Comment (gnalh) [inline script.sh:32]: Thoughts on looking in a few places for this? I suggest we look at the top directory of the repo, the current directory where this script is run, and in a `tools` directory (if it exists).

Comment (riya-n) [reply script.sh:32]: Yeah I think it makes sense to look for this in a couple of places

Comment (riya-n) [reply script.sh:32]: @gnalh Something like this? Or do we want to be more specific about where to search
```bash
trunk_analytics_uploader_path=$(find ./ -name trunk-analytics-uploader)
set -x
if [[ ! (-f ${trunk_analytics_uploader_path}) ]]; then
    curl -fsSL --retry 3 "https://trunk.io/releases/analytics-cli/latest" -o ./trunk-analytics-uploader
    trunk_analytics_uploader_path=./trunk-analytics-uploader
fi
chmod +x "${trunk_analytics_uploader_path}"
set +x

echo "${trunk_analytics_uploader_path}"

${trunk_analytics_uploader_path} upload \
    --junit-paths "${JUNIT_PATHS}" \
    --org-url-slug "${ORG_URL_SLUG}" \
    --token "${TOKEN}" \
    --tags "${TAGS}"

rm -rf "${trunk_analytics_uploader_path}"
```

Comment (gnalh) [reply script.sh:32]: I think it's preferable to be less verbose and only look in specific places. `find ./ -name trunk-analytics-uploader` would be slow in large repos.

Comment (riya-n) [reply script.sh:32]: As discussed in planning, we can have this for now and then follow up with more optimizations later. Will keep the ticket open: https://linear.app/trunk/issue/TRUNK-10226/cli-action-detect-if-binary-already-present-in-the-local-fs-and-skip

**Review by gnalh — APPROVED** [review]

**Pull request merged**